components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90

```diff
@@ -1669,24 +1673,34 @@ subroutine mam_amicphys_1subarea_cloudy(       &
       if ( ( do_cond                         ) .and. &
            ( gaexch_h2so4_uptake_optaa == 2 ) ) then
          do igas = 1, ngas
-            if ((igas == igas_h2so4) .or. (igas == igas_nh3)) then
-! if gaexch_h2so4_uptake_optaa == 2, then
-!    if qgas increases from pre-gaschem to post-cldchem,
-!       start from the pre-gaschem mix-ratio and add in the production
-!       during the integration
-!    if it decreases, 
-!       start from post-cldchem mix-ratio
-! *** currently just do this for h2so4 and nh3
+!! if gaexch_h2so4_uptake_optaa == 2, then
+!!    if qgas increases from pre-gaschem to post-cldchem,
+!!       start from the pre-gaschem mix-ratio and add in the production
+!!       during the integration
+!!    if it decreases, 
+!!       start from post-cldchem mix-ratio
+!! *** currently just do this for h2so4 and nh3
+            do ispec = 1, nso4
+            if (igas == igas_h2so4(ispec)) then
```

Comment:
Same block for these two logicals, we can revert to original more compact version.